ccmlib/scylla_repository.py

```diff
@@ -0,0 +1,170 @@
+        cdir = version_directory(version)
+        scylla_java_tools_dir = os.path.join(cdir, 'scylla-java-tools')
+        for jar_file in glob.glob(scylla_java_tools_dir + "/*.jar"):
+            shutil.copy(jar_file, os.path.join(scylla_java_tools_dir, "lib"))
+
+        url = os.environ.get('SCYLLA_JMX_PACKAGE', os.path.join(s3_url, 'scylla-jmx-package.tar.gz'))
+        download_version(os.path.join(version, 'jmx'), verbose=verbose, url=url)
+
+        cdir = version_directory(version)
+    return cdir, version
+
+
+min_attributes = ('scheme', 'netloc')
+
+
+def is_valid(url, qualifying=None):
+    qualifying = min_attributes if qualifying is None else qualifying
+    token = urllib.parse.urlparse(url)
+    return all([getattr(token, qualifying_attr)
+                for qualifying_attr in qualifying])
+
+
+def download_version(version, url=None, verbose=False):
+    """Download, extract, and build Cassandra tarball.
+
+    if binary == True, download precompiled tarball, otherwise build from source tarball.
+    """
+    target_dir = None
+    try:
+        if os.path.exists(url) and url.endswith('.tar.gz'):
+            target = url
+        elif is_valid(url):
+            _, target = tempfile.mkstemp(suffix=".tar.gz", prefix="ccm-")
+            __download(url, target, show_progress=verbose)
+        else:
+            raise ArgumentError("unsupported url={}".format(url))
+
+        if verbose:
+            print_("Extracting %s as version %s ..." % (target, version))
+        tar = tarfile.open(target)
+        tar.extractall(path=os.path.join(__get_dir(), version))
+        tar.close()
+
+    except urllib.error.URLError as e:
+        msg = "Invalid version %s" % version if url is None else "Invalid url %s" % url
+        msg = msg + " (underlying error is: %s)" % str(e)
+        raise ArgumentError(msg)
+    except tarfile.ReadError as e:
+        raise ArgumentError("Unable to uncompress downloaded file: %s" % str(e))
+    except CCMError as e:
+        if target_dir:
+            # wipe out the directory if anything goes wrong. Otherwise we will assume it has been compiled the next time it runs.
+            try:
+                rmdirs(target_dir)
+                print_("Deleted %s due to error" % target_dir)
+            except:
+                raise CCMError("Building C* version %s failed. Attempted to delete %s but failed. This will need to be manually deleted" % (version, target_dir))
+        raise e
+
+
+def directory_name(version):
+    return os.path.join(__get_dir(), version)
+
+
+def version_directory(version):
+    dir = directory_name(version)
+    if os.path.exists(dir):
+        try:
+            validate_install_dir(dir)
+            return dir
+        except ArgumentError:
+            rmdirs(dir)
+            return None
+    else:
+        return None
+
+
+def clean_all():
+    rmdirs(__get_dir())
+
+
+def __download(url, target, username=None, password=None, show_progress=False):
```

Comment:
Check if we can import this from the cassandra file

Reply:
fixed